### PR TITLE
#213 EA-1873 added rights field to transcriptions body.

### PR DESCRIPTION
**Pull request by rgraf**
Implemented rights filed handling for transcriptions body. Created JUnit test.

### CODE REVIEW

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 1]
```diff
@@ -829,6 +855,11 @@ public void validateWebAnnotation(Annotation webAnnotation) throws ParamValidati
 		case TAGGING:
 			validateTag(webAnnotation);
 			break;
+		case TRANSCRIBING:
+			// if rights are provided, check if it belongs to the valid license list
+			if (webAnnotation != null && webAnnotation.getBody() != null && webAnnotation.getBody().getRights() != null)
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: The if statement ca nbe included in the validate method

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 2]
```diff
@@ -588,6 +590,30 @@ protected boolean validateResource(String url) throws ParamValidationException {
...
+	 * @return true if provided right is in a list of valid licenses
+	 * @throws ParamValidationException
+	 */
+	protected boolean validateRights(String right) throws ParamValidationException {
+
+		// remove version from the right
+		String RIGHT_DELIMITER = "/";		
+		String[] rightElems = right.split(RIGHT_DELIMITER);
+		String[] normalizedRightElems = Arrays.copyOf(rightElems, rightElems.length-1);
+		String normalizedRight = String.join(RIGHT_DELIMITER, normalizedRightElems)+RIGHT_DELIMITER;
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: use index of an substring .... parse and rejoin is not an efficient implementation

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 3]
```diff
@@ -588,6 +590,30 @@ protected boolean validateResource(String url) throws ParamValidationException {
...
+		String transcriptionLicenses = getConfiguration().getTranscriptionsLicenses();
+		List<String> licensesList = Arrays.asList(transcriptionLicenses.split("\\s*,\\s*"));
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: use StringUtils.split(string, ',')

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 4]
```diff
@@ -588,6 +590,30 @@ protected boolean validateResource(String url) throws ParamValidationException {
...
+		Set<String> rights = new HashSet<String>(licensesList);
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: Use Streams and laizy loading. create getAcceptedLicenceses method. 

Set<String> set = Stream.of("a", "b")
                        .collect(Collectors.toCollection(HashSet::new));

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 1 above)
Comment (gsergiu) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: also the name of the right field was changed to edmRights see https://europeana.atlassian.net/browse/EA-1873

Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: if statement moved to validate method

Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: rights Field changed to edmRights

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 3 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: regular split replaced by StringUtils.split

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 2 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: parsing and rejoin method will be raplaced by more efficient method

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 4 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: stream and collection methods were implemented

**Pull request merged**